crates/goose/src/mcp_server_runner.rs

```diff
@@ -0,0 +1,45 @@
+use anyhow::{anyhow, Result};
+use goose_mcp::{
+    AutoVisualiserRouter, ComputerControllerServer, DeveloperServer, MemoryServer, TutorialServer,
+};
+use rmcp::{transport::stdio, ServiceExt};
+
+/// Run an MCP server by name
+///
+/// This function handles the common logic for starting MCP servers.
+/// The caller is responsible for setting up logging before calling this function.
+pub async fn run_mcp_server(name: &str) -> Result<()> {
+    if name == "googledrive" || name == "google_drive" {
+        return Err(anyhow!(
+            "the built-in Google Drive extension has been removed"
+        ));
+    }
+
+    tracing::info!("Starting MCP server");
+
+    match name {
```

Comment:
we were missing this previously?

Reply:
ah this is the replacement for run_server in the old mcp crate.

Reply:
There were two copies of similar files, both with slightly different logic. One in `cli` and one in `server`. I am just deduping and improving here, but I should find a final home for it that keeps the dependency direction good. Will make an adjustment.